File: update_colors.py
import os
import re

def walk_sync(dir_path):
    file_list = []
    for root, dirs, files in os.walk(dir_path):
        for file in files:
            if file.endswith('.css') or file.endswith('.js') or file.endswith('.html'):
                file_list.append(os.path.join(root, file))
    return file_list

files = walk_sync('./public')
for file in files:
    with open(file, 'r', encoding='utf-8') as f:
        content = f.read()
    
    changed = False

    # Replace colors
    if 'rgba(108,99,255' in content:
        content = content.replace('rgba(108,99,255', 'rgba(0,184,148')
        changed = True
    
    if '#6C63FF' in content or '#6c63ff' in content:
        content = re.sub(r'(?i)#6C63FF', '#00B894', content)
        changed = True

    if '#5A52E0' in content or '#5a52e0' in content:
        content = re.sub(r'(?i)#5A52E0', '#008F72', content)
        changed = True

    # Remove :root hardcoded variables from feature css files
    if 'whiteboard.css' in file or 'files.css' in file or 'timer.css' in file:
        if ':root{' in content or ':root {' in content:
            content = re.sub(r':root\s*\{[^}]+\}', '', content)
            changed = True

    # Replace #1a1a24 in whiteboard files
    if 'whiteboard.js' in file or 'whiteboard.css' in file:
        if '#1a1a24' in content:
            content = content.replace('#1a1a24', '#FFFFFF')
            changed = True

    # Include base.css in whiteboard/index.html
    if 'whiteboard/index.html' in file:
        if 'base.css' not in content:
            content = content.replace('<link rel="stylesheet" href="/features/whiteboard/whiteboard.css"/>', '<link rel="stylesheet" href="/core/css/base.css"/>\n<link rel="stylesheet" href="/features/whiteboard/whiteboard.css"/>')
            changed = True

    if changed:
        with open(file, 'w', encoding='utf-8') as f:
            f.write(content)
        print('Updated:', file)

File: update_colors.js
const fs = require('fs');
const path = require('path');

const walkSync = (dir, filelist = []) => {
  fs.readdirSync(dir).forEach(file => {
    const dirFile = path.join(dir, file);
    if (fs.statSync(dirFile).isDirectory()) {
      filelist = walkSync(dirFile, filelist);
    } else {
      if (dirFile.endsWith('.css') || dirFile.endsWith('.js') || dirFile.endsWith('.html')) {
        filelist.push(dirFile);
      }
    }
  });
  return filelist;
};

const files = walkSync('./public');
files.forEach(file => {
  let content = fs.readFileSync(file, 'utf8');
  let changed = false;

  // Replace colors
  if (content.includes('rgba(108,99,255')) {
    content = content.replace(/rgba\(108,99,255/g, 'rgba(0,184,148');
    changed = true;
  }
  if (content.includes('#6C63FF')) {
    content = content.replace(/#6C63FF/ig, '#00B894');
    changed = true;
  }
  if (content.includes('#5A52E0')) {
    content = content.replace(/#5A52E0/ig, '#008F72');
    changed = true;
  }
  
  // Remove :root hardcoded variables from feature css files
  if (file.includes('whiteboard.css') || file.includes('files.css') || file.includes('timer.css')) {
    if (content.includes(':root{')) {
      content = content.replace(/:root\{[^}]+\}/s, '');
      changed = true;
    }
  }

  // Also replace #1a1a24 in whiteboard.js to #FFFFFF for canvas bg
  if (file.includes('whiteboard.js')) {
    if (content.includes('#1a1a24')) {
      content = content.replace(/#1a1a24/g, '#FFFFFF');
      changed = true;
    }
  }

  // Replace #1a1a24 in whiteboard.css 
  if (file.includes('whiteboard.css')) {
    if (content.includes('#1a1a24')) {
      content = content.replace(/#1a1a24/g, '#FFFFFF');
      changed = true;
    }
  }

  // Include base.css in whiteboard/index.html
  if (file.includes('whiteboard/index.html')) {
    if (!content.includes('base.css')) {
      content = content.replace('<link rel="stylesheet" href="/features/whiteboard/whiteboard.css"/>', '<link rel="stylesheet" href="/core/css/base.css"/>\n<link rel="stylesheet" href="/features/whiteboard/whiteboard.css"/>');
      changed = true;
    }
  }

  if (changed) {
    fs.writeFileSync(file, content);
    console.log('Updated:', file);
  }
});
